### PR TITLE
Updated plugins.md

### PR DESCRIPTION
**Pull request by DielN**
Added info on new plugin functions:
- app.glib_rename
- app.getToolInfo
- app.setCurrentLayerName
- app.setBackgroundName
- app.scaleTextElements
- app.getDisplayDpi

I also corrected app.changeBackgroundPdfPageNr function name (was app.changePdfBackgroundPageNr) and reordered the function list to match the order in luapi_application.h.
The broken/outdated links to the plugin API have been updated, too.
Unified plugin descriptions to end with no comma/dot. I'm considering unifying the wording of the description, e.g. some begin with for and/or use the -ing form, and some simply use the simple present (sets, scrolls), which is also what I used for the plugin descriptions.

Still looking over the rest of the plugins page, so this is WIP. Feedback is welcome.

### CODE REVIEW

**Comment by DielN** [comment]
I just noticed the getToolInfo function has only recently been pushed to master and is not yet available in the latest release.

Should this be removed for now? Also, should we link to master for the luapi_application.h or to the most recent release branch (1.1 in this case)?

**Comment by Technius** [comment]
Thanks for helping out!

> Also, should we link to master for the luapi_application.h or to the most recent release branch 

Most recent release is preferred, but new plugin API functions should link to a pinned nightly/master commit.

> Should it be removed for now?

It can be added, but it needs to be accompanied by a note saying that it's only available in the nightly builds.

File: docs/guide/plugins/plugins.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ The Lua code file referenced in the `plugin.ini` file must define an `initUi`-fu
...
-- the `accelerator` used to execute the plugin callback function. The format looks like `<Control>a` or `<Shift><Alt>F1` or `<Release>z` (for key release). See [GTK3 reference](https://developer.gnome.org/gtk3/stable/gtk3-Keyboard-Accelerators.html#gtk-accelerator-parse) for details and note that release key accelerators currently do not work, see [this issue](https://github.com/xournalpp/xournalpp/issues/2396). The accelerator can be omitted. Using accelerators without modifiers like (`<Alt>`, `<Control>`, `<Shift>`) is allowed, but will likely break the text tool.
+- the `accelerator` used to execute the plugin callback function. The format looks like `<Control>a` or `<Shift><Alt>F1` or `<Release>z` (for key release). See [GTK3 reference](https://docs.gtk.org/gtk3/func.accelerator_parse.html) for details and note that release key accelerators currently do not work, see [this issue](https://github.com/xournalpp/xournalpp/issues/2396). The accelerator can be omitted. Using accelerators without modifiers like (`<Alt>`, `<Control>`, `<Shift>`) is allowed, but will likely break the text tool.
```
Comment (DielN) [inline docs/guide/plugins/plugins.md:55]: GTK3 documentation on accelerators seems to be broken due to the tags being parsed incorrectly. I opened an issue on their GitLab but I'm not sure how fast they fix it. The [GTK4 docs](https://docs.gtk.org/gtk4/func.accelerator_parse.html) seem to be fine, would that be an option?

**Comment by DielN** [comment]
I'm pretty much done with updating things. I am thinking about adding a more detailed description for getToolInfo like getDocumentStructure has and maybe for changeCurrentPageBackground to explain the possible config values, not sure about that. Would that fit into the same pull request or should I create another one for that?

Apart from that, I'd say the page is ready for review. I hope the changed wording of the descriptions is fine, any feedback is appreciated.

A summary of what I updated:
- updated broken/outdated links (but gtk3 docs linked are broken, see review above)
- added description of new API functions, for the function getToolInfo only available in nightly I put a note about it not being in the latest (non-nightly) release yet
- unified the wording of the descriptions for better looks and for better readability
- changed some descriptions to include more info on what the function's capable of or what it does (saveAs, registerUi, changeCurrentPageBackground)
- fixed changeBackgroundPdfPageNr function name being misspelled
- added a link to the plugins folder in the repository for examples of plugins to look at

File: docs/guide/plugins/plugins.md (hunk 1 above)
Comment (rolandlo) [reply docs/guide/plugins/plugins.md:55]: Yes, the GTK4 docs look fine. Interestingly they don't mention a Release key any more (and I think it never got implemented, reading through the code file that does the parsing). So `<Release>z` shouldn't be mentioned at all, issue [#2396](https://github.com/xournalpp/xournalpp/issues/2396) should be closed and not mentioned.

File: docs/guide/plugins/plugins.md [hunk 2]
```diff
@@ -78,32 +78,40 @@ end
...
-- `app.uiAction` for simulating a toolbar/menubar click starting some action,
-- `app.uiActionSelected` for notifying action listeners about selected options,
-- `app.changeCurrentPageBackground` for changing the background of the current page,
-- `app.sidebarAction` for actions accessible in the sidebar
-- `app.layerAction` for actions accessible in the layer controller
-- `app.changeToolColor` for changing the color of any tool with color capabilities,
-- `app.changePdfBackgroundPageNr` to change the pdf background page number
-- `app.getDocumentStructure` to get lots of useful info on the document, also used for applying operations on all/selected pages
-- `app.scrollToPage`scrolls relatively or absolutely to a page
+- `app.msgbox` shows a message box to the user and lets them choose an option by clicking a button
```
Comment (rolandlo) [inline docs/guide/plugins/plugins.md:90]: The grammar should be fixed somehow. Not sure what is the best way.
```suggestion
- `app.msgbox` shows a message box to the user and lets the user choose an option by clicking a button
```

**Review by rolandlo — CHANGES_REQUESTED** [review]
Thank you for your efforts. In general I like your updates and changes and have only marked two minor things to change.

Please open a new PR for adding a more detailed description for getToolInfo and/or changeCurrentPageBackground

Perhaps I will have time to finish https://github.com/xournalpp/xournalpp/pull/3566 by then, so the `app.export` command could also be added.

File: docs/guide/plugins/plugins.md (hunk 2 above)
Comment (DielN) [reply docs/guide/plugins/plugins.md:90]: How about this:
```suggestion
- `app.msgbox` displays a message box with the specified buttons and returns the button clicked by the user
```

or simply leave out the first mention of the user (who else would the msgbox be shown to?), I think "displays" fits better then:
```suggestion
- `app.msgbox` displays a message box and lets the user choose an option by clicking a button
```

File: docs/guide/plugins/plugins.md (hunk 1 above)
Comment (DielN) [reply docs/guide/plugins/plugins.md]: I hope I understood you correctly. It links to the GTK4 docs now and the release-accelerator isn't mentioned anymore, is that alright?

File: docs/guide/plugins/plugins.md (hunk 2 above)
Comment (rolandlo) [reply docs/guide/plugins/plugins.md:90]: I like the first suggestion. It looks more precise than the latter one.

File: docs/guide/plugins/plugins.md (hunk 1 above)
Comment (rolandlo) [reply docs/guide/plugins/plugins.md]: Yes, that's fine.

**Comment by rolandlo** [comment]
Can you please squash your commits into one?

**Comment by rolandlo** [comment]
Merging in 24 hours if no objections are raised.

**Comment by rolandlo** [comment]
Thanks again!

**Pull request merged**